test/distributed/fsdp/test_fsdp_misc.py

```diff
@@ -489,6 +499,91 @@ def __init__(self, rank):
                 fsdp, process_group=self.process_group, assert_fn=self.assertEqual
             )
 
+    @skip_if_lt_x_gpu(2)
+    def test_change_wrapped_module_after_ctor(self):
+        """
+        Tests changing an FSDP instance's wrapped module after the FSDP
+        constructor.
+        """
+        dist.set_debug_level(dist.DebugLevel.DETAIL)
+
+        class Model(nn.Module):
+            def __init__(self) -> None:
+                super().__init__()
+                self.seq1 = nn.Sequential(
+                    nn.Linear(5, 5),
+                    nn.Linear(5, 5),
+                )
+                self.seq2 = nn.Sequential(nn.Linear(5, 5))
+                self.lin = nn.Linear(5, 5)
+                self.relu = nn.ReLU()
+
+            def forward(self, x: torch.Tensor) -> torch.Tensor:
+                return self.lin(self.relu(self.seq2(self.relu(self.seq1(x)))))
+
+        def get_fsdp_model():
+            fsdp_kwargs = {"use_orig_params": False}
+            model = Model().cuda()
+            model.seq1 = FSDP(model.seq1, **fsdp_kwargs)
+            model.seq2[0] = FSDP(model.seq2[0], **fsdp_kwargs)
+            model = FSDP(model, **fsdp_kwargs)
+            return model
+
+        # Wrap with `CheckpointWrapper` *after* FSDP construction
+        model = get_fsdp_model()
+        non_reentrant_wrapper = functools.partial(
+            checkpoint_wrapper,
+            offload_to_cpu=False,
+            checkpoint_impl=CheckpointImpl.NO_REENTRANT,
+        )
+        check_fn = lambda submodule: isinstance(submodule, nn.Linear)
```

Comment:
lambda should be anonymous

Reply:
Could you explain what the difference between anonymous and lambda is?